src/Components/DataHistoryExplorer/Internal/TimeSeriesViewer/TimeSeriesViewer.types.ts

```diff
@@ -1,14 +1,38 @@
 import {
     IPivotStyles,
+    ISpinnerStyles,
     IStyle,
     IStyleFunctionOrObject,
     ITheme
 } from '@fluentui/react';
+import { ADXTimeSeries, IComponentError } from '../../../../Models/Constants';
 import { IDataHistoryTimeSeriesTwin } from '../../../../Models/Constants/Interfaces';
+import { IllustrationMessageStyles } from '../../../IllustrationMessage/IllustrationMessage.types';
+import { IDataHistoryExplorerChartOptions } from '../../DataHistoryExplorer.types';
+import { ITimeSeriesChartStyles } from './Internal/TimeSeriesChart/TimeSeriesChart.types';
+import {
+    ITimeSeriesCommandBarOptions,
+    ITimeSeriesCommandBarStyles,
+    IViewerModeProps
+} from './Internal/TimeSeriesCommandBar/TimeSeriesCommandBar.types';
+import {
+    ITimeSeriesTableStyles,
+    TimeSeriesTableRow
+} from './Internal/TimeSeriesTable/TimeSeriesTable.types';
 
+export interface TimerSeriesViewerData {
+    chart: Array<ADXTimeSeries>;
+    table: Array<TimeSeriesTableRow>;
+}
 export interface ITimeSeriesViewerProps {
-    timeSeriesTwinList: Array<IDataHistoryTimeSeriesTwin>;
-    onMissingSeriesData?: (seriesIds: Array<string>) => void;
+    timeSeriesTwins: Array<IDataHistoryTimeSeriesTwin>;
+    data: TimerSeriesViewerData;
+    isLoading: boolean;
+    viewerModeProps: IViewerModeProps;
```

Comment:
IViewerModeProps is also a field for ITimeSeriesCommandBarProps which is props of the child here. So since it is not fully overlapping in terms of types (ITimeSeriesCommandBarOptions is not same as IDataHistoryExplorerChartOptions), I am going to leave it as is here.